sections/security/bcryptpasswords.md

```diff
@@ -0,0 +1,28 @@
+# Avoid using the Node.js Crypto library for passwords, use Bcrypt
+
+### One Paragraph Explainer
+When storing user passwords, using an adaptive hashing algorithm such as bcrypt, offered by the [bcrypt npm module](https://www.npmjs.com/package/bcrypt) is recommended as opposed to using the native Node.js Crypto module. `Math.random()` should also never be used for as part of any password or token generation due to it's predictability.
+
+The `bcrypt` module or similar should be used as opposed to the JavaScript implementation, as when using `bcrypt`, a number of 'rounds' can be specified in order to provide a secure hash. This sets the work factor or the number of 'rounds' the data is processed for, and more hashing rounds leads to more secure hash (although this at the cost of CPU time). The introduction of hashing rounds means that the brute force factor is significantly reduced, as password crackers are slowed down increasing the time required to generate one attempt.
+
+### Code example
```

Comment:
@i0natan @js-kyle maybe we should create and enforce a guideline for JavaScript code examples, because we'll end up with many different types of code snippets, using different ECMAScript features and so on.

Reply:
@BrunoScheufler what for example might confuse the user? I need an example of confusing examples or any existing examples that are not consistent

Reply:
@i0natan it's not so much about confusion, rather inconsistency, for example [this function](https://github.com/i0natan/nodebestpractices/pull/166/files#diff-ccb88acb49767f79fac21b4425a5e551R17) uses a function literal, whereas we also use arrow functions in other cases. I think it's about the things that might look a bit nitpicky but make a difference viewed as a whole.

Reply:
@BrunoScheufler see your point, I'm just no sure how "structured and unified" we wanna be. Where would you maintain these guidelines? won't this make our writing guidelines too long and maybe, as you said, nitty gritty? I absolutely agree that it's better to have the code examples unified, just not sure whether we should be so strict